pkgs/development/python-modules/openai-triton/0000-strip-and-replace-ptxas.patch

```diff
@@ -0,0 +1,57 @@
+diff --git a/python/setup.py b/python/setup.py
+index 2ac3accd2..f26161c72 100644
+--- a/python/setup.py
++++ b/python/setup.py
+@@ -101,25 +101,6 @@ def get_thirdparty_packages(triton_cache_path):
+ # ---- package data ---
+ 
+ 
+-def download_and_copy_ptxas():
+-    base_dir = os.path.dirname(__file__)
+-    src_path = "bin/ptxas"
+-    url = "https://conda.anaconda.org/nvidia/label/cuda-12.0.0/linux-64/cuda-nvcc-12.0.76-0.tar.bz2"
+-    dst_prefix = os.path.join(base_dir, "triton")
+-    dst_suffix = os.path.join("third_party", "cuda", src_path)
+-    dst_path = os.path.join(dst_prefix, dst_suffix)
+-    if not os.path.exists(dst_path):
+-        print(f'downloading and extracting {url} ...')
+-        ftpstream = urllib.request.urlopen(url)
+-        file = tarfile.open(fileobj=ftpstream, mode="r|*")
+-        with tempfile.TemporaryDirectory() as temp_dir:
+-            file.extractall(path=temp_dir)
+-            src_path = os.path.join(temp_dir, src_path)
+-            os.makedirs(os.path.split(dst_path)[0], exist_ok=True)
+-            shutil.copy(src_path, dst_path)
+-    return dst_suffix
+-
+-
+ # ---- cmake extension ----
+ 
+ 
+@@ -200,8 +181,6 @@ class CMakeBuild(build_ext):
+         subprocess.check_call(["cmake", "--build", "."] + build_args, cwd=self.build_temp)
+ 
+ 
+-download_and_copy_ptxas()
+-
+ setup(
+     name="triton",
+     version="2.0.0",
+diff --git a/python/triton/runtime/jit.py b/python/triton/runtime/jit.py
+index 496cea311..f8dd48345 100644
+--- a/python/triton/runtime/jit.py
++++ b/python/triton/runtime/jit.py
+@@ -90,12 +90,7 @@ def version_key():
+     for lib in pkgutil.iter_modules([language_path]):
+         with open(lib.module_finder.find_spec(lib.name).origin, "rb") as f:
+             contents += [hashlib.md5(f.read()).hexdigest()]
+-    # ptxas version
+-    try:
+-        ptxas_version = hashlib.md5(subprocess.check_output(["ptxas", "--version"])).hexdigest()
+-    except Exception:
+-        ptxas_version = ''
+-    return '-'.join(triton.__version__) + '-' + ptxas_version + '-' + '-'.join(contents)
++    return '-'.join(triton.__version__) + '-@ptxas_version@' + '-' + '-'.join(contents)
```

Comment:
What I mean in https://github.com/NixOS/nixpkgs/pull/263048/files#r1372394031 is that maybe we can just remove this part of the patch? AFAIU, if you don't install ptxas, then this code will just put an empty string in place of the ptxas version, and that already distinguishes the cuda-free installation from any other installation? (I'm presuming they use this for versioning their cache or something. What do I know though)

Reply:
Yes, but the project itself forces a ptxas download, so we shouldn't expect that try block to ever get to except.
I'm more surprised it doesn't outright error.

Reply:
>  forces a ptxas download when building

That we patch out

Reply:
I don't know how many projects check this hash, but what I'm worried about is a project that does check for the ptxas hash even though it doesn't use it for a non-cuda build.
Do you think we should patch those other projects when/if they come up instead?

Reply:
I haven't seen a single such project, but sure, if we encounter one we can patch it. And I wouldn't worry about the "ptxas hash affecting non-cuda builds" because the authors expressly approved of making ptxas optional (but that they wouldn't work on that themselves)

Reply:
I may actually just be able to update this to ``2.1.0``, which is hidden on the ``2.1.x`` branch.

Reply:
I'll just remove that part of the patch and work on a more comprehensive change on another PR, along with torch, which I think I'm going to split up like we do with blender.